### DISCUSSION TITLE
Issue while using large Anonymous block

### DISCUSSION THREAD

**Discussion by ankitchandura** [oracle/python-oracledb#37] [Q&A]
While using large Oracle PL/SQL anonymous block ( **around 20,147 characters** ). I've saved this block inside one independent .sql file. This .sql content I'm assigning to one variable and that variable I'm using in:

`mCmd = "cursor.execute(vQuery1)"`
`exec(mCmd)`

While executing the .py file I'm receiving below error:

`Traceback (most recent call last):`
`  File "/home/chanduraankit/deliverable/devl/bin/./test.py", line 83, in <module>`
`    exec(mCmd)`
`  File "<string>", line 1, in <module>`
`cx_Oracle.DatabaseError: ORA-00972: identifier is too long`

Anonymous PL/SQL block is executing properly on Oracle Session using Oracle tool.

Can you please help identifying what is the issue here?

Thanks,
Ankit.

**Comment by anthony-tuininga** [comment]
I'm not sure why you are using `exec()` instead of just executing the code directly -- but that ought to be a side issue. In addition, you appear to be using cx_Oracle and this is the python-oracledb repository! Regardless, can you share more detail on platforms, versions, contents of the PL/SQL block, etc.?

**Reply by ankitchandura** [reply]
Hi Antony,
I'm using below technologies:

Unix/Linux: `GNU/Linux : 3.10.0-1160.45.1.el7.x86_64`
Database: `Oracle Database 19c Enterprise Edition Release 19.0.0.0.0 - Production Version 19.13.0.0.0`
Contents in PL/SQL block: `Declare
   Variable declaration;
   Cursor to fetch some date;
Begin
   Variable initialization;
   Insert into Log_Table;
   Business Logic;
   Dynamic SQL for Create Table;
   Update Log_Table
   Insert Into some_table;
exception <exception_name>
   exception handling;
end;`

Does this suffice? Please let me know if more detail is required.

Thanks.

**Reply by anthony-tuininga** [reply]
Can you supply a complete runnable test case that demonstrates the problem? The PL/SQL block you provided isn't valid code -- so hard to say what you are running into exactly!
